### PR TITLE
Improve search performance in `FileBrowser`

### LINKED ISSUE #6874: Program freeze when searching in "My Computer"
### Bug Summary
I have a second samples folder on my E drive, and when I went to search it using the "My Computer" tab, the program froze. This may have something to do with some recent push that seems to make all the folders in the default sample folder open automatically when searched, if I were to hazard a guess, but I could be wrong.
#### Steps to reproduce
Go into the "My Computer" sidebar tab, and search.
#### Expected behavior
Not freezing the program
#### Actual behavior
It freezes
#### Affected LMMS versions
image
image

### PR DESCRIPTION
**Pull request by sakertooth**
Improves the search performance of the file browser by delegating the search to a worker thread. The main thread then builds the tree and displays it to the user.

Fixes #6874.

### CODE REVIEW

**Comment by Spekular** [comment]
> Clicking on a file that can be previewed will play the file in its entirely and not require to hold the mouse.

FYI this was already an option via #5787, see also #5427. I'd very much like to keep the option to stop samples on mouse release.

**Comment by sakertooth** [comment]
> FYI this was already an option via https://github.com/LMMS/lmms/pull/5787, see also https://github.com/LMMS/lmms/pull/5427. I'd very much like to keep the option to stop samples on mouse release.

Thanks for the heads up. Seems like I overrode that PR a bit. I'll fix it and push the commits soon.

**Comment by sakertooth** [comment]
Realized that this PR was far too invasive for little benefit other than search performance, so I put the focus on just that rather than a lot of unnecessary changes.

File: src/gui/FileBrowser.cpp [hunk 1]
```diff
@@ -1276,5 +1454,18 @@ QString FileItem::extension(const QString & file )
...
+QString FileItem::defaultFilters()
+{
+	// TODO: Supported extensions should be in a centralized location
+	auto simpleExtensions
+		= QString{"*.mmp *.mpt *.mmpz *.xpf *.xml *.xiz *.sf2 *.sf3 *.pat *.mid *.midi *.rmi *.dll *.lv2"};
+#ifdef LMMS_BUILD_LINUX
+	simpleExtensions += " *.so";
+#endif
+	auto audioExtensions = QString{"*.wav *.ogg *.ds *.flac *.spx *.voc *.aif *.aiff *.au *.raw *.wav *.ogg *.ds "
+								   "*.flac *.spx *.voc *.aif *.aiff *.au *.raw"};
```
Comment (zonkmachine) [inline src/gui/FileBrowser.cpp:1466]: mp3 just disappeard.  ;)

PS. why are some of the suffixes duplicated? Is this by design?

Comment (zonkmachine) [reply src/gui/FileBrowser.cpp:1466]: Actually, it's not some of the suffixes. The string is duplicated so they all appear twice. Surely this is wrong? They show up fine where I've tested it though (afp).

Comment (sakertooth) [reply src/gui/FileBrowser.cpp:1466]: I should've waited for a review, but didn't know when that was going to happen and needed this ASAP. I’ll send in a fix immediately.

Comment (sakertooth) [reply src/gui/FileBrowser.cpp:1466]: Fixed in #6978.

Comment (zonkmachine) [reply src/gui/FileBrowser.cpp:1466]: Great!

**Pull request merged**